src/main/java/cz/cvut/kbss/termit/persistence/dao/LuceneSearchDao.java

```diff
@@ -19,41 +19,67 @@
 
 import cz.cvut.kbss.jopa.model.EntityManager;
 import cz.cvut.kbss.jopa.model.query.Query;
+import cz.cvut.kbss.jopa.vocabulary.SKOS;
+import cz.cvut.kbss.termit.dto.search.FacetedSearchResult;
 import cz.cvut.kbss.termit.dto.search.FullTextSearchResult;
+import cz.cvut.kbss.termit.dto.search.SearchParam;
 import cz.cvut.kbss.termit.util.Constants;
+import cz.cvut.kbss.termit.util.Utils;
 import cz.cvut.kbss.termit.util.Vocabulary;
 import jakarta.annotation.Nonnull;
 import jakarta.annotation.Nullable;
+import jakarta.annotation.PostConstruct;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.context.annotation.Profile;
+import org.springframework.data.domain.Pageable;
 import org.springframework.stereotype.Repository;
 
 import java.net.URI;
+import java.util.Collection;
 import java.util.Collections;
 import java.util.List;
 import java.util.Objects;
+import java.util.stream.Collectors;
 
 /**
- * {@link SearchDao} extension for Lucene-based repositories. These support rich search strings with wildcards and
+ * Search data access object using Lucene-based repositories. These support rich search strings with wildcards and
  * operators.
  * <p>
  * This DAO automatically adds a wildcard to the last token in the search string, so that results for incomplete words
  * are returned as well.
  */
 @Repository
-@Profile("lucene")  // Corresponds to a profile set in pom.xml
-public class LuceneSearchDao extends SearchDao {
+public class LuceneSearchDao {
```

Comment:
Can we call it `SearchDao` now? I don't think it is necessary to indicate it being based on Lucene indexes with class name (now that there is no alternative from which we would need to differentiate it).